### PR TITLE
Backup/Restore, blob separation, purge function 

### PR DESCRIPTION
**Pull request by Chrislevi**
- Backup and Restore procedures added. 3.2.1 newest feature
- Single blob for each supported format. 
- Added purge ability
- Minor indention fix

### CODE REVIEW

**Comment by Chrislevi** [comment]
@hagzag

**Comment by Chrislevi** [comment]
@hagzag @samherve Bump.

File: defaults/main.yml [hunk 1]
```diff
@@ -104,7 +131,7 @@ nexus_repos_maven_proxy:
 nexus_repos_maven_hosted:
   - name: private-release
     version_policy: release
-    write_policy: allow_once
+    write_policy: allow
```
Comment (samherve) [inline defaults/main.yml]: Why not keeping allow_once on maven releases ?

File: defaults/main.yml [hunk 2]
```diff
@@ -128,15 +162,15 @@ _nexus_privilege_defaults:
     - read
 
 _nexus_repos_maven_defaults:
-  blob_store: default # Note : cannot be updated once the repo has been created
+  blob_store: "{{ nexus_blob_prefix }}mvn"
```
Comment (samherve) [inline defaults/main.yml]: I think that there may be an issue about a kind of "backward compatibility" here : if someone has already provisionned a nexus instance using the (previous) default role vars, then this update of the role becomes incompatible with his current nexus installation ?

File: tasks/nexus_install.yml [hunk 3]
```diff
@@ -18,6 +29,8 @@
     src: "{{ nexus_download_dir }}/{{ nexus_package }}"
     dest: "{{ nexus_installation_dir }}"
     creates: "{{ nexus_installation_dir }}/nexus-{{ nexus_version }}"
+    owner: "{{ nexus_os_user }}"
```
Comment (samherve) [inline tasks/nexus_install.yml]: Why should the nexus executable be allowed to overwrite itself ? I considered this a security point to have the program (as opposed to data) filed owned by root

**Comment by hagzag** [comment]
@samherve could you take a look @ this PR this is one of my colleagues which added backup & restore capabilities

**Review by samherve — CHANGES_REQUESTED** [review]

File: defaults/main.yml (hunk 1 above)
Comment (Chrislevi) [reply defaults/main.yml]: I'll revert it to allow_once.

File: defaults/main.yml (hunk 2 above)
Comment (Chrislevi) [reply defaults/main.yml]: I'm working on a 'splited' and default flows to the blobstore configuration

File: tasks/nexus_install.yml (hunk 3 above)
Comment (Chrislevi) [reply tasks/nexus_install.yml]: Rolling back, had problems deploying scripts to the script dir located in nexus_install dir. never mind, i've found a simple solution.

**Comment by Chrislevi** [comment]
@samherve  Please re-overview the pr, i've pushed the requested changes :)

**Comment by Chrislevi** [comment]
bump @samherve

File: tasks/nexus_install.yml [hunk 4]
```diff
@@ -61,7 +77,8 @@
       - "{{ nexus_installation_dir }}/nexus-latest/etc/jetty"
       - "{{ nexus_installation_dir }}/nexus-latest/etc/fabric"
       - "{{ nexus_installation_dir }}/nexus-latest/etc/logback"
-  when: nexus_version >= '3.1.0'
+      - "{{ nexus_installation_dir }}/nexus-latest/etc/scripts"
+  when: nexus_version >= '3.2.1-01'
```
Comment (samherve) [inline tasks/nexus_install.yml:81]: I think this line should stay as it was `nexus_version >= '3.1.0'` ? there was a change in the config paths at version 3.1.0, so there is no need to bump this version (see line 66 above).

**Review by samherve — CHANGES_REQUESTED** [review]
Hi, I finally found time to test this review, apart from y single comment about keeping `when: nexus_version >= '3.1.0'` , everything works fine to me. 

I also tested it succesfully with the latest 3.3.0-01 nexus release.

**Pull request merged**